app/controllers/lottery_controller.rb

```diff
@@ -78,21 +117,21 @@ def create_new_teams_for_bidding_response(teams, assignment, priority_info)
     end
   end
 
-  def merge_bids_from_different_previous_teams(assignment, team_id, user_ids, priority_info)
-    # Select data from `priority_info` variable that only related to team members in current team and transpose it.
-    # For example, below matrix shows 4 topics (key) and correponding priorities given by 3 team members (value).
+  def merge_bids_from_different_previous_teams(assignment, team_id, user_ids, user_bidding_info)
+    sign_up_topics = assignment.sign_up_topics
+
+    # Select data from `user_bidding_info` variable that only related to team members in current team and transpose it.
+    # For example, below matrix shows 4 topics (key) and corresponding priorities given by 3 team members (value).
     # {
     #   1: [1, 2, 3],
     #   2: [0, 1, 2],
     #   3: [2, 3, 1],
     #   4: [2, 0, 1]
     # }
-    bidding_matrix = {}
-    priority_info.each do |info|
-      next unless user_ids.include? info[:pid]
-      bids = info[:ranks]
-      assignment.sign_up_topics.each_with_index do |topic, index|
-        bidding_matrix[topic.id] = [] unless bidding_matrix[topic.id]
+    bidding_matrix = Hash.new([])
+    current_team_members_info = user_bidding_info.select {|info| user_ids.include? info[:pid] }
```

Comment:
In what case would an entry in user_bidding_info not include :pid key? Is this line necessary?